Release Notes 2.7.11 (#8382)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## Release 2.7.11

### Exciting New Features 🎉

> - Add refactor of results into a standalone artifact table — https://github.com/PrefectHQ/prefect/pull/8164

Reviewer: ```suggestion
- Add refactor of results into a standalone `artifact` table — https://github.com/PrefectHQ/prefect/pull/8164
```
Maybe this should go in experimental?

Reviewer 2: Agreed this should go in experimental.

> - Add Prefect Cloud Events schema and clients. — https://github.com/PrefectHQ/prefect/pull/8357

Reviewer 2: @chrisguidry how should we announce this? Does this make sense to place in experimental for now or do we want to make a splash or... ?

Reviewer 3: Yes, at most, let's say "Experimental".  We don't have it hooked up to the main Client yet.  I'd say we can either tag it experimental or quietly drop/de-emphasize it.  @WillRaphaelson what do you think?

Reviewer 4: Sorry thanks for catching this guidry, yes experimental please!

### Enhancements
- Update pinned pip version in Docker images — https://github.com/PrefectHQ/prefect/pull/8289
- Update default task run name to exclude hash of task key — https://github.com/PrefectHQ/prefect/pull/8292

> - Add functionality that updates preinstalled packages in Docker images — https://github.com/PrefectHQ/prefect/pull/8288

Reviewer 2: "Add functionality" is weird here, perhaps?

- Update Docker images to update preinstalled packages on build

> - Add `PREFECT_LOGGING_ORION_WHEN_MISSING_FLOW` setting to handle error raised when logger called outside of flow run context — https://github.com/PrefectHQ/prefect/pull/8311

Reviewer 2: I think this is more like "Add `PREFECT_LOGGING_ORION_WHEN_MISSING_FLOW` to allow loggers to be used outside of flows" — could also elevate this into the top-level with an example.

### Fixes
- Fix cancellation bug — https://github.com/PrefectHQ/prefect/pull/8315
- Fix Kubernetes job watch exit when no timeout given — https://github.com/PrefectHQ/prefect/pull/8350

> - Fix default value for priority on `WorkQueue` core schema — https://github.com/PrefectHQ/prefect/pull/8373

Reviewer 2: What does this mean if I'm a user? This adds a default value for priority for compatibility with previous client versions, right?

Reviewer 5: We should move this to experimental

> - Fix UI to use `recentFlowRunFilter` instead of `flowRunFilter` — https://github.com/PrefectHQ/prefect/pull/8380

Reviewer 2: What does this mean if I'm a user?

Reviewer 5: We can just remove this. It was part of work pool development

### Experimental
- Add ability to exclude experimental fields in API calls — https://github.com/PrefectHQ/prefect/pull/8274
- Add work queue data migration — https://github.com/PrefectHQ/prefect/pull/8327

> - Add clean up migration — https://github.com/PrefectHQ/prefect/pull/8331

Reviewer 2: These can probably be collapsed? You can link multiple pull requests for a single item

- Add updates to work queue `last_polled` time when polling work pools — https://github.com/PrefectHQ/prefect/pull/8338
- Add CLI support for work pools — https://github.com/PrefectHQ/prefect/pull/8259
- Add fields to `work_queue` table to accommodate work pools — https://github.com/PrefectHQ/prefect/pull/8264

### Documentation
- Add git commands to Prefect Recipes contribution page — https://github.com/PrefectHQ/prefect/pull/8283
- Add `retry_delay_seconds` and `exponential_backoff` examples to Tasks retries documentation — https://github.com/PrefectHQ/prefect/pull/8280
- Add role permissions regarding block secrets — https://github.com/PrefectHQ/prefect/pull/8309
- Add getting started tutorial video to Prefect Cloud Quickstart — https://github.com/PrefectHQ/prefect/pull/8336
- Add tips for re-registering blocks from Prefect Collections — https://github.com/PrefectHQ/prefect/pull/8333

> - Add examples and corrections for Kubernetes infrastructure overrides — https://github.com/PrefectHQ/prefect/pull/8312

Reviewer: ```suggestion
- Improve examples for Kubernetes infrastructure overrides — https://github.com/PrefectHQ/prefect/pull/8312
```

# Prefect Release Notes

## Release 2.7.11

### Enhancements
- Update pinned pip version in Docker images — https://github.com/PrefectHQ/prefect/pull/8289
- Update default task run name to exclude hash of task key — https://github.com/PrefectHQ/prefect/pull/8292
- Add functionality that updates preinstalled packages in Docker images — https://github.com/PrefectHQ/prefect/pull/8288
- Add `PREFECT_LOGGING_ORION_WHEN_MISSING_FLOW` setting to handle error raised when logger called outside of flow run context — https://github.com/PrefectHQ/prefect/pull/8311

### Fixes

> - Fix cancellation bug — https://github.com/PrefectHQ/prefect/pull/8315

Reviewer 5: ```suggestion
- Fix bug preventing agents from properly updating Cancelling runs to a Cancelled state — https://github.com/PrefectHQ/prefect/pull/8315
```

- Fix cancellation bug — https://github.com/PrefectHQ/prefect/pull/8315
- Fix Kubernetes job watch exit when no timeout given — https://github.com/PrefectHQ/prefect/pull/8350
- Fix default value for priority on `WorkQueue` core schema — https://github.com/PrefectHQ/prefect/pull/8373
- Fix UI to use `recentFlowRunFilter` instead of `flowRunFilter` — https://github.com/PrefectHQ/prefect/pull/8380

### Experimental

> - We're working on work pools, groups of work queues. Together, work pools & queues give you greater flexibility and control in organizing and prioritizing work.

Reviewer 5: ```suggestion
- We're working on work pools, groups of work queues. Together, work pools & queues give you greater flexibility and control to organize and prioritize work.
```

# Prefect Release Notes

## Release 2.7.11

### Work pools are here!

> We're excited to announce the release of work pools, a new feature that allows you to create logical groups for your work queues. Work pools are a great way to organize your work queues and ensure that your most important work is always prioritized. You can read more about work pools in our [documentation](https://docs.prefect.io/concepts/work-pools/).

Reviewer 5: ```suggestion
We're excited to announce the release of work pools, groups of work queues. Together, work pools & queues give you greater flexibility and control in organizing and prioritizing work. You can read more about work pools in our [documentation](https://docs.prefect.io/concepts/work-pools/).
```

> We're excited to announce the release of work pools, a new feature that allows you to create logical groups for your work queues. Work pools are a great way to organize your work queues and ensure that your most important work is always prioritized. You can read more about work pools in our [documentation](https://docs.prefect.io/concepts/work-pools/).

Reviewer 5: ```suggestion
```

### Enhancements

> - Add refactor of results into a standalone artifact table — https://github.com/PrefectHQ/prefect/pull/8164

Reviewer 5: ```suggestion
```

Reviewer 5: I think we should remove this since its not user facing.

> - Add Prefect Cloud Events schema and clients. — https://github.com/PrefectHQ/prefect/pull/8357

Reviewer 5: ```suggestion
- Add Prefect Cloud Events schema and clients — https://github.com/PrefectHQ/prefect/pull/8357
```

- Update pinned pip version in Docker images — https://github.com/PrefectHQ/prefect/pull/8289

> - Update default task run name to exclude hash of task key — https://github.com/PrefectHQ/prefect/pull/8292

Reviewer 5: ```suggestion
- Update default task run name to exclude the task key hash — https://github.com/PrefectHQ/prefect/pull/8292
```

Reviewer 2: Well... it's a hash of the "task key" there is no "task key hash" but this reads nicely too.

Reviewer 5: haha fair I'll drop it

- Add functionality that updates preinstalled packages in Docker images — https://github.com/PrefectHQ/prefect/pull/8288
- Add `PREFECT_LOGGING_ORION_WHEN_MISSING_FLOW` setting to handle error raised when logger called outside of flow run context — https://github.com/PrefectHQ/prefect/pull/8311

### Fixes
- Fix cancellation bug — https://github.com/PrefectHQ/prefect/pull/8315
- Fix Kubernetes job watch exit when no timeout given — https://github.com/PrefectHQ/prefect/pull/8350
- Fix default value for priority on `WorkQueue` core schema — https://github.com/PrefectHQ/prefect/pull/8373
- Fix UI to use `recentFlowRunFilter` instead of `flowRunFilter` — https://github.com/PrefectHQ/prefect/pull/8380

### Experimental

> - Add ability to exclude experimental fields in API calls — https://github.com/PrefectHQ/prefect/pull/8274

Reviewer 5: ```suggestion
- We're working on work pools, groups of work queues. Together, work pools & queues give you greater flexibility and control in organizing and prioritizing work.
- Add ability to exclude experimental fields in API calls — https://github.com/PrefectHQ/prefect/pull/8274
```
